discovery-handlers/opcua/src/discovery_impl.rs

```diff
@@ -417,4 +446,22 @@ mod tests {
         );
         assert!(discovery_urls.is_empty());
     }
+
+    #[test]
+    // Test that it converts the discovery url to an ip address if the discovery url is a hostname that is not resolvable
+    fn test_get_discovery_url_ip() {
+        let ip_url = "opc.tcp://192.168.0.1:50000/";
+        let discovery_url = "opc.tcp://OPCTest:50000/OPCUA/Simluation";
```

Comment:
From a DNS point of view this could one day become resolvable (unlikely but yet), we may want to use the reserved tld `.invalid` here to ensure this (and make it clearer at first glance this is a not resolvable domain).
```suggestion
        let discovery_url = "opc.tcp://OPCTest.invalid:50000/OPCUA/Simluation";
```